artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageSubscriptionCounterImpl.java

```diff
@@ -374,8 +403,43 @@ private ItemOper(PageSubscriptionCounterImpl counter, long id, int add) {
       @Override
       public void afterCommit(Transaction tx) {
          for (ItemOper oper : operations) {
-            oper.counter.incrementProcessed(oper.id, oper.amount);
+            oper.counter.incrementProcessed(oper.id, oper.amount, oper.size);
          }
       }
    }
+
+   private static class PendingCounter {
+      final long id;
+      final AtomicInteger count;
```

Comment:
could this be an int, with a static atomic updater (ditto with the atomic below also)

Reply:
What do you mean by atomic updater, do you mean using something like AtomicIntegerFieldUpdater ?

Reply:
I'm about to push an updated PR using the field updater classes.

Reply:
Yes this i what i meant , ill have a look at the updated PR in a bit/tomorrow.